### PR TITLE
more clear names for methods of Space

### PR DESCRIPTION
**Pull request by zach-nervana**
rename `Space.val_matches_space_definition` -> `contains`
rename `Space.is_point_in_space_shape` -> `valid_index`

### CODE REVIEW

**Comment by zach-nervana** [comment]
This is intended to address issue #180

File: rl_coach/spaces.py [hunk 1]
```diff
@@ -117,9 +117,10 @@ def high(self, val: Union[None, int, float, np.ndarray]):
             if type(self._high) == int or type(self._high) == float:
                 self._high = np.ones(self.shape)*self._high
 
-    def val_matches_space_definition(self, val: Union[int, float, np.ndarray]) -> bool:
+    def contains(self, val: Union[int, float, np.ndarray]) -> bool:
```
Comment (galnov) [inline rl_coach/spaces.py]: 1. Looks fine. Consider changing to is_valid_index()
2. Please update the documentation files as well (both docs_raw and the compiled version under docs) with the new names.

**Comment by zach-nervana** [comment]
@galnov your requested changes have been made

**Comment by zach-nervana** [comment]
This fixes issue #180

**Pull request merged**